### PR TITLE
Feat: userDetail-follow-unfollow

### PR DESCRIPTION
**Pull request by lomesshh**
- In this pr I've added: 
-- edit user profile ( with profile pic )
-- follow user / unfollow user

### CODE REVIEW

File: src/frontend/pages/auth/Profile.jsx [hunk 1]
```diff
@@ -7,26 +7,36 @@ import { Createpost } from "frontend/styled-component/homeStyled";
...
   const dispatch = useDispatch();
+  const [userData, setUserData] = useState({ ...user });
 
   useEffect(() => {
+    dispatch(fetchUserData(user._id));
     dispatch(getUsersPost(user.username));
-  }, [allPosts]);
+  }, [user]);
+
+  console.log("ssssas", user);
```
Comment (romabulani) [inline src/frontend/pages/auth/Profile.jsx]: Remove this console.log

File: src/frontend/services/UserService.jsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+export const fetchUserData = (userId) => {
+  return async (dispatch) => {
+    try {
+      const res = await axios.get(`/api/users/${userId}`);
+      dispatch(getUserData(res.data.user));
+    } catch (error) {
+      console.log(error);
+    }
+  };
```
Comment (romabulani) [inline src/frontend/services/UserService.jsx:29]: In all catch statements, try to use `console.error`

**Review by romabulani — APPROVED** [review]
- Modular and well structured code
- Functionality is working as expected
- Good to merge

**Review by shrey27 — APPROVED** [review]
Code is clean and well structured
UI is working as expected
PR is good to merge

File: src/frontend/services/UserService.jsx (hunk 2 above)
Comment (lomesshh) [reply src/frontend/services/UserService.jsx:29]: ok, ill change all catch statements to console.error in my next PR

File: src/frontend/pages/auth/Profile.jsx (hunk 1 above)
Comment (lomesshh) [reply src/frontend/pages/auth/Profile.jsx]: Done

**Pull request merged**